test/units/modules/network/aci/test_aci_rest.py

```diff
@@ -0,0 +1,299 @@
+        self.assertEqual(json_result, xml_result)
+
+    def test_empty_response(self):
+        self.maxDiffi = None
+
+        if sys.version_info < (3, 0):
+            expected_json_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as JSON, see 'raw' output. No JSON object could be decoded",
+                'raw': '',
+            }
+
+        else:
+            expected_json_result = {
+                u'error_code': -1,
+                u'error_text': u"Unable to parse output as JSON, see 'raw' output. Expecting value: line 1 column 1 (char 0)",
+                u'raw': u'',
+            }
+
+        json_response = ''
+        json_result = aci_response(json_response, 'json')
+        self.assertEqual(expected_json_result, json_result)
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
+            return
+
+        elif etree.LXML_VERSION < (3, 3, 0, 0):
+            expected_xml_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as XML, see 'raw' output. None",
+                'raw': '',
+            }
+
+        elif sys.version_info < (3, 0):
+            expected_xml_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as XML, see 'raw' output. None (line 0)",
+                'raw': '',
+            }
+
+        else:
+            expected_xml_result = {
+                u'error_code': -1,
+                u'error_text': u"Unable to parse output as XML, see 'raw' output. None (line 0)",
+                u'raw': u'',
+            }
+
+        xml_response = ''
+        xml_result = aci_response(xml_response, 'xml')
+        self.assertEqual(expected_xml_result, xml_result)
+
+    def test_invalid_response(self):
+        self.maxDiff = None
+
+        if sys.version_info < (2, 7):
+            expected_json_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as JSON, see 'raw' output. Expecting object: line 1 column 8 (char 8)",
+                'raw': '{ "aaa":',
+            }
+
+        elif sys.version_info < (3, 0):
+            expected_json_result = {
+                'error_code': -1,
+                'error_text': "Unable to parse output as JSON, see 'raw' output. No JSON object could be decoded",
+                'raw': '{ "aaa":',
+            }
+
+        else:
+            expected_json_result = {
+                u'error_code': -1,
+                u'error_text': u"Unable to parse output as JSON, see 'raw' output. Expecting value: line 1 column 9 (char 8)",
+                u'raw': u'{ "aaa":',
+            }
+
+        json_response = '{ "aaa":'
+        json_result = aci_response(json_response, 'json')
+        self.assertEqual(expected_json_result, json_result)
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
```

Comment:
Another test that I would move into version specific classes with pytest skipifs